### ISSUE TITLE
Update solr spatial filter method

### ISSUE THREAD

**Issue by cole-fields** [genkimaps/gis-hub#121]
Related Issue [#109]. 

__Purpose:__ Improve the solr spatial filter feature on GIS Hub. The default method sent to solr is 'Intersects'. 
__Problem:__ With the Intersects method, too many results are returned. This is because we are using the convex hull geom to intersect the user-drawn bounding box. 
__Solution:__ 
- [ ] Add a setting in the config file that defines the spatial filter type ('Within', 'Intersects', or 'Contains')
- [ ] Update before_search method in plugin.py
- [ ] Copy/modify `_params_for_solr_spatial_field_search` method to plugin.py from https://github.com/ckan/ckanext-spatial/blob/5a987828cb8e76f4904c6f23542a593255171a94/ckanext/spatial/plugin/__init__.py#L318
- [ ] Update docs with new method. I think Within is the most intuitive.

**Comment by cole-fields** [comment]
@genkimaps This is now merged with master. I think it makes the spatial search behave a lot more intuitively, and greatly reduces the number of returned datasets. Returns datasets whose convex hulls are within the bounding box drawn by user rather than the default 'Intersects' method.

**Comment by genkimaps** [comment]
@cole-fields  Yes I think that's definitely a better option.  Thanks for digging into the details to figure this out.
